File: app/src/main/java/com/nguyendevs/stkh/Manager/UtilityManager.java
package com.nguyendevs.stkh.Manager;

import android.content.ClipData;
import android.content.ClipboardManager;
import android.content.Context;
import android.content.Intent;
import android.net.Uri;
import android.widget.EditText;
import android.widget.Toast;

public class UtilityManager {
    private final Context context;
    private final EditText txtResult;

    public UtilityManager(Context context, EditText txtResult) {
        this.context = context;
        this.txtResult = txtResult;
    }
    public void copyToClipboard(String text) {
        if (text != null && !text.isEmpty()) {
            ClipboardManager clipboard = (ClipboardManager) context.getSystemService(Context.CLIPBOARD_SERVICE);
            ClipData clip = ClipData.newPlainText("Copied Text", text);
            clipboard.setPrimaryClip(clip);
            Toast.makeText(context, "Đã sao chép vào clipboard!", Toast.LENGTH_SHORT).show();
        } else {
            Toast.makeText(context, "Không có nội dung để sao chép!", Toast.LENGTH_SHORT).show();
        }
    }
    public void shareText() {
        String text = txtResult.getText().toString();
        if (!text.isEmpty()) {
            Intent shareIntent = new Intent(Intent.ACTION_SEND);
            shareIntent.setType("text/plain");
            shareIntent.putExtra(Intent.EXTRA_TEXT, text);
            context.startActivity(Intent.createChooser(shareIntent, "Chia sẻ qua"));
        } else {
            Toast.makeText(context, "Không có nội dung để chia sẻ!", Toast.LENGTH_SHORT).show();
        }
    }
    public void searchOnGoogle() {
        String query = txtResult.getText().toString().trim();
        if (!query.isEmpty()) {
            if (query.contains(".") && !query.startsWith("http://") && !query.startsWith("https://")) {
                query = "https://" + query;
            } else if (!query.startsWith("http://") && !query.startsWith("https://")) {
                query = "https://www.google.com/search?q=" + Uri.encode(query);
            }
            Intent intent = new Intent(Intent.ACTION_VIEW, Uri.parse(query));
            context.startActivity(intent);
        } else {
            Toast.makeText(context, "Không có nội dung để tìm kiếm!", Toast.LENGTH_SHORT).show();
        }
    }
}

File: app/src/main/java/com/nguyendevs/stkh/manager/UtilityManager.kt
package com.nguyendevs.stkh.manager

import android.content.ClipData
import android.content.ClipboardManager
import android.content.Context
import android.content.Intent
import android.net.Uri
import android.widget.EditText
import com.nguyendevs.stkh.util.showToast

class UtilityManager(
    private val context: Context,
    private val txtResult: EditText
) {

    fun copyToClipboard(text: String) {
        if (text.isNotEmpty()) {
            val clipboard = context.getSystemService(Context.CLIPBOARD_SERVICE) as ClipboardManager
            val clip = ClipData.newPlainText("Copied Text", text)
            clipboard.setPrimaryClip(clip)
            context.showToast("Đã sao chép vào clipboard!")
        } else {
            context.showToast("Không có nội dung để sao chép!")
        }
    }

    fun shareText() {
        val text = txtResult.text.toString()
        if (text.isNotEmpty()) {
            val shareIntent = Intent(Intent.ACTION_SEND).apply {
                type = "text/plain"
                putExtra(Intent.EXTRA_TEXT, text)
            }
            context.startActivity(Intent.createChooser(shareIntent, "Chia sẻ qua"))
        } else {
            context.showToast("Không có nội dung để chia sẻ!")
        }
    }

    fun searchOnGoogle() {
        var query = txtResult.text.toString().trim()
        if (query.isNotEmpty()) {
            query = when {
                query.startsWith("http://") || query.startsWith("https://") -> query
                query.contains(".") -> "https://$query"
                else -> "https://www.google.com/search?q=${Uri.encode(query)}"
            }
            val intent = Intent(Intent.ACTION_VIEW, Uri.parse(query))
            context.startActivity(intent)
        } else {
            context.showToast("Không có nội dung để tìm kiếm!")
        }
    }
}
